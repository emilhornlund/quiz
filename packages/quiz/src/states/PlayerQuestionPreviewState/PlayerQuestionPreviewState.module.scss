@import '../../styles/colors';
@import '../../styles/helpers';

.main {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  height: 100%;
  flex: 1;

  .title {
    display: flex;
    flex-direction: row;
    justify-content: center;
    font-weight: bolder;
    text-align: center;
    color: white;

    @include mobile {
      width: 100%;
      line-height: 3.5rem;

      @include fontSize(46px);
    }

    @include tablet {
      width: $mobile-width;
      line-height: 5rem;

      @include fontSize(66px);
    }

    @include desktop {
      width: $tablet-width;
      line-height: 6rem;

      @include fontSize(76px);
    }
  }
}