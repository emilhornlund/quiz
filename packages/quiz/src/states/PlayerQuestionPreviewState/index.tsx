export type { PlayerQuestionPreviewStateProps } from './PlayerQuestionPreviewState'
export { default } from './PlayerQuestionPreviewState'
