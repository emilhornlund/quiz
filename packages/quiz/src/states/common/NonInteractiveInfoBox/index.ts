export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export { default } from './NonInteractiveInfoBox'
