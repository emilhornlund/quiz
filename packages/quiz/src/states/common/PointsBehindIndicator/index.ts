export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export { default } from './PointsBehindIndicator'
