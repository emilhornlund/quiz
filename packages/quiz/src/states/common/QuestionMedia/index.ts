export type { QuestionMediaProps } from './QuestionMedia'
export { default } from './QuestionMedia'
