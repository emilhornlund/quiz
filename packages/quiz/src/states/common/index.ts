export type { NonInteractiveInfoBoxProps } from './NonInteractiveInfoBox'
export type { PointBehindIndicatorProps } from './PointsBehindIndicator'
export type { QuestionAnswerPickerProps } from './QuestionAnswerPicker'
export type { QuestionMediaProps } from './QuestionMedia'
export type { QuestionTextPreviewProps } from './QuestionTextPreview'
export type { QuestionTypePointsBarProps } from './QuestionTypePointsBar'

export { default as GamePage } from './GamePage'
export { default as NonInteractiveInfoBox } from './NonInteractiveInfoBox'
export { default as PointsBehindIndicator } from './PointsBehindIndicator'
export { default as QuestionAnswerPicker } from './QuestionAnswerPicker'
export { default as QuestionMedia } from './QuestionMedia'
export { default as QuestionTextPreview } from './QuestionTextPreview'
export { default as QuestionTypePointsBar } from './QuestionTypePointsBar'
