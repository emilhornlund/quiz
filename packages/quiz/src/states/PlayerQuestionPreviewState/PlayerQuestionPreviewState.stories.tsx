import { GameEventType, QuestionType } from '@quiz/common'
import type { Meta, StoryObj } from '@storybook/react'
import { withRouter } from 'storybook-addon-remix-react-router'

import PlayerQuestionPreviewState from './PlayerQuestionPreviewState'

const meta = {
  component: PlayerQuestionPreviewState,
  decorators: [withRouter],
  parameters: {
    layout: 'fullscreen',
  },
} satisfies Meta<typeof PlayerQuestionPreviewState>

export default meta
type Story = StoryObj<typeof meta>

export const Default = {
  args: {
    event: {
      type: GameEventType.GameQuestionPreviewPlayer,
      player: {
        nickname: 'FrostyBear',
        score: 10458,
      },
      question: {
        type: QuestionType.MultiChoice,
        question: 'Who painted The Starry Night?',
      },
      countdown: {
        expiryTime: new Date(Date.now() + 60 * 1000).toISOString(),
        serverTime: new Date().toISOString(),
      },
      pagination: { current: 1, total: 20 },
    },
  },
} satisfies Story
