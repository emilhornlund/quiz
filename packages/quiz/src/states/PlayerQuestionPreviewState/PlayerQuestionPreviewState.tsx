import { GameQuestionPreviewPlayerEvent } from '@quiz/common'
import React, { FC } from 'react'

import { PlayerGameFooter, ProgressBar, Typography } from '../../components'
import { GamePage } from '../common'

export interface PlayerQuestionPreviewStateProps {
  event: GameQuestionPreviewPlayerEvent
}

const PlayerQuestionPreviewState: FC<PlayerQuestionPreviewStateProps> = ({
  event: {
    player: { nickname: playerNickname, score: playerTotalScore },
    question: { type: questionType, question: questionValue },
    countdown,
    pagination: { current: currentQuestion, total: totalQuestions },
  },
}) => (
  <GamePage
    align="space-between"
    footer={
      <PlayerGameFooter
        currentQuestion={currentQuestion}
        totalQuestions={totalQuestions}
        nickname={playerNickname}
        totalScore={playerTotalScore}
      />
    }>
    <div>{questionType}</div>
    <Typography variant="subtitle" size="medium">
      {questionValue}
    </Typography>
    <ProgressBar countdown={countdown} />
  </GamePage>
)

export default PlayerQuestionPreviewState
