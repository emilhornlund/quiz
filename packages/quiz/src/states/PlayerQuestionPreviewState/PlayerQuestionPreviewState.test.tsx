import { GameEventType, QuestionType } from '@quiz/common'
import { render } from '@testing-library/react'
import React from 'react'
import { describe, expect, it } from 'vitest'

import PlayerQuestionPreviewState from './PlayerQuestionPreviewState'

describe('PlayerQuestionPreviewState', () => {
  it('should render PlayerQuestionPreviewState', () => {
    const { container } = render(
      <PlayerQuestionPreviewState
        event={{
          type: GameEventType.GameQuestionPreviewPlayer,
          player: {
            nickname: 'FrostyBear',
            score: 10458,
          },
          question: {
            type: QuestionType.Multi,
            question: 'Who painted The Starry Night?',
          },
          progress: {
            value: 0.75,
          },
          pagination: { current: 1, total: 20 },
        }}
      />,
    )

    expect(container).toMatchSnapshot()
  })
})
