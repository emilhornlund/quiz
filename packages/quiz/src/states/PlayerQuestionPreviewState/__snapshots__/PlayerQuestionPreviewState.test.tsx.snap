// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PlayerQuestionPreviewState > should render PlayerQuestionPreviewState 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <button
        class="logo"
      >
        <img
          alt="Klurigo"
          class="icon"
          src="/src/assets/images/klurigo-icon.svg"
        />
        <span
          class="text"
        >
          Klurigo
        </span>
      </button>
      <div
        class="side"
      >
        <div
          class="buttonInputContainer buttonInputKindPrimary buttonInputSizeSmall"
        >
          <button
            data-testid="test-fullscreen-button-button"
            id="fullscreen-button"
            name="fullscreen-button"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-maximize "
              data-icon="maximize"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M200 32L56 32C42.7 32 32 42.7 32 56l0 144c0 9.7 5.8 18.5 14.8 22.2s19.3 1.7 26.2-5.2l40-40 79 79-79 79L73 295c-6.9-6.9-17.2-8.9-26.2-5.2S32 302.3 32 312l0 144c0 13.3 10.7 24 24 24l144 0c9.7 0 18.5-5.8 22.2-14.8s1.7-19.3-5.2-26.2l-40-40 79-79 79 79-40 40c-6.9 6.9-8.9 17.2-5.2 26.2s12.5 14.8 22.2 14.8l144 0c13.3 0 24-10.7 24-24l0-144c0-9.7-5.8-18.5-14.8-22.2s-19.3-1.7-26.2 5.2l-40 40-79-79 79-79 40 40c6.9 6.9 17.2 8.9 26.2 5.2s14.8-12.5 14.8-22.2l0-144c0-13.3-10.7-24-24-24L312 32c-9.7 0-18.5 5.8-22.2 14.8s-1.7 19.3 5.2 26.2l40 40-79 79-79-79 40-40c6.9-6.9 8.9-17.2 5.2-26.2S209.7 32 200 32z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      class="content fullWidth spaceBetweenAlign"
    >
      <div>
        MULTI_CHOICE
      </div>
      <div
        class="typography subtitle medium"
      >
        Who painted The Starry Night?
      </div>
      <div
        class="progressBar"
      >
        <span
          style="width: 100%; transition: width 0.1s linear;"
        />
      </div>
    </div>
    <div
      class="footer"
    >
      <div
        class="main"
      >
        <div
          class="questions"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-question "
            data-icon="circle-question"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 512A256 256 0 1 0 256 0a256 256 0 1 0 0 512zM169.8 165.3c7.9-22.3 29.1-37.3 52.8-37.3l58.3 0c34.9 0 63.1 28.3 63.1 63.1c0 22.6-12.1 43.5-31.7 54.8L280 264.4c-.2 13-10.9 23.6-24 23.6c-13.3 0-24-10.7-24-24l0-13.5c0-8.6 4.6-16.5 12.1-20.8l44.3-25.4c4.7-2.7 7.6-7.7 7.6-13.1c0-8.4-6.8-15.1-15.1-15.1l-58.3 0c-3.4 0-6.4 2.1-7.5 5.3l-.4 1.2c-4.4 12.5-18.2 19-30.6 14.6s-19-18.2-14.6-30.6l.4-1.2zM224 352a32 32 0 1 1 64 0 32 32 0 1 1 -64 0z"
              fill="currentColor"
            />
          </svg>
          <span>
            1
             / 
            20
          </span>
        </div>
        <div
          class="nickname"
        >
          <span>
            FrostyBear
          </span>
        </div>
        <div
          class="score"
        >
          <span>
            10458
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
